crates/compute/src/layer.rs

```diff
@@ -448,6 +454,72 @@ pub trait ComputeLayer<F: Field>: 'static + Sync {
 	) -> Result<(), Error>;
 }
 
+/// An interface for constructing execution kernels.
+///
+/// A _kernel_ is a program that executes synchronously in one thread, with access to
+/// local memory buffers.
+///
+/// See [`ComputeLayer::accumulate_kernels`] for more information.
+pub trait KernelBuilder<F> {
```

Comment:
I'm only confused about the terminology consistency. As for be it's pretty weird to see a builder, that doesn't have any build methods

Reply:
Yeah, that's fair. I'm ambivalent about Executor vs Builder as far as terminology. In the IC1 context, builder makes sense because it's building a description of a kernel program, which will then be compiled and executed. In the CPU context, it executes immediately.

@BenjaminTrapani I assume you're OK if we name the trait KernelExecutor?

Reply:
Renaming to `KernelExecutor` sounds good to me. This component is called `KernelBuilder` in ic1 because it defines the kernel layout and compiler input for a transformation corresponding to the chunk size. The built artifacts consist of the compiled instructions, input / output layouts and kernel-local memory layout.